referenceframe/frame_json.go

```diff
@@ -1,117 +1,86 @@
 package referenceframe
 
 import (
-	"encoding/json"
-	"fmt"
-
-	"github.com/golang/geo/r3"
-	"github.com/mitchellh/mapstructure"
-
+	"reflect"
 	spatial "go.viam.com/rdk/spatialmath"
-	"go.viam.com/rdk/utils"
 )
 
-// FrameMapConfig represents the format for configuring a Frame object.
-type FrameMapConfig map[string]interface{}
+// LinkConfig is a StaticFrame that also has a specified parent
+type LinkConfig struct {
+	ID          string                    `json:"id"`
+	Translation spatial.TranslationConfig `json:"translation"`
+	Orientation *spatial.OrientationConfig `json:"orientation"`
+	Geometry    *spatial.GeometryConfig    `json:"geometry,omitempty"`
+	Parent      string                    `json:"parent,omitempty"`
+}
 
-// UnmarshalFrameJSON deserialized json into a reference referenceframe.
-func UnmarshalFrameJSON(data []byte) (Frame, error) {
-	config := FrameMapConfig{}
-	err := json.Unmarshal(data, &config)
-	if err != nil {
-		return nil, err
-	}
+type JointCfg struct {
```

Comment:
Nit: Lint will complain about docs for these, but DATA-790 should bring in comments once you rebase here.